### PR TITLE
DM-40520: Add support for binning images to ip_isr

### PR DESCRIPTION
**Pull request by czwa**
Add method for ip_isr to produce binned images.

### CODE REVIEW

**Review by PaulPrice — APPROVED** [review]

**Review by PaulPrice — APPROVED** [review]

File: python/lsst/ip/isr/isrTask.py [hunk 1]
```diff
@@ -917,6 +935,25 @@ class IsrTaskConfig(pipeBase.PipelineTaskConfig,
...
+    doBinnedExposures = pexConfig.Field(
+        dtype=bool,
+        doc="Should binned exposures be calculated?",
+        default=False,
+    )
+    binFactor1 = pexConfig.Field(
+        dtype=int,
+        doc="Binning factor for first binned exposure.",
+        default=8,
+        check=lambda x: x >= 0,
```
[the remark this answers is not in the document]
Comment (czwa) [reply python/lsst/ip/isr/isrTask.py:948]: Changed to be `> 1`.  You're right that zero makes zero sense, and one also isn't useful (as it just duplicates the regular output).

File: python/lsst/ip/isr/isrTask.py [hunk 2]
```diff
@@ -2559,6 +2605,28 @@ def flatContext(self, exp, flat, dark=None):
...
+        Returns
+        -------
+        bin1 : `lsst.afw.image.Exposure`
+            Binned exposure using binFactor1.
+        bin2 : `lsst.afw.image.Exposure`
+            Binned exposure using binFactor2.
+        """
+        mi = exposure.getMaskedImage()
+
+        bin1 = afwMath.binImage(mi.clone(), self.config.binFactor1)
```
[the remark this answers is not in the document]
Comment (czwa) [reply python/lsst/ip/isr/isrTask.py:2625]: I somehow convinced myself that the function operated in place, which I can see is not the case from looking in afw.  Removed.

File: python/lsst/ip/isr/isrTask.py [hunk 3]
```diff
@@ -260,6 +260,19 @@ class IsrTaskConnections(pipeBase.PipelineTaskConnections,
...
     )
+    outputBin1Exposure = cT.Output(
+        name="postIsrBin1",
+        doc="First binned image.",
+        storageClass="ExposureF",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+    outputBin2Exposure = cT.Output(
+        name="postIsrBin2",
+        doc="First binned image.",
```
[the remark this answers is not in the document]
Comment (plazas) [reply python/lsst/ip/isr/isrTask.py:271]: Sorry, why do you need to produce two binned images?

[the remark this answers is not in the document]
Comment (PaulPrice) [reply python/lsst/ip/isr/isrTask.py]: We found in Pan-STARRS that it's useful to have a low-resolution version for use in tables of exposures, and a medium resolution version for inspection when you notice a problem.

[the remark this answers is not in the document]
Comment (plazas) [reply python/lsst/ip/isr/isrTask.py]: I see, thanks! Would it be worth it to add a comment regarding this, for future readers of the code who have the same question?

**Pull request merged**